app-backend/db/src/main/scala/TaskDao.scala

```diff
@@ -263,14 +263,25 @@ object TaskDao extends Dao[Task] {
       user: User
   ): ConnectionIO[Int] = {
     for {
-      geomO <- taskGridFeatureCreate.geometry match {
-        case Some(g) => Option(g).pure[ConnectionIO]
-        case None =>
-          AnnotationProjectDao.getFootprint(
-            taskProperties.annotationProjectId
-          )
+      annotationProjectO <- AnnotationProjectDao.getById(
+        taskProperties.annotationProjectId
+      )
+      geomO <- (taskGridFeatureCreate.geometry, annotationProjectO) match {
+        case (Some(g), _) => Option(g).pure[ConnectionIO]
+        case (_, Some(annotationProject)) =>
+          AnnotationProjectDao.getFootprint(annotationProject.id)
+        case _ => None.pure[ConnectionIO]
+      }
+      taskSizeO = (
```

Comment:
This whole block can be:

```suggestion
      taskSizeO = taskGridFeatureCreate.properties.sizeMeters orElse (annotationProjectO flatMap { _.taskSizeMeters })
```
```